src/compiler/transformers/test/transpile.ts

```diff
@@ -1,16 +1,24 @@
 import type * as d from '@stencil/core/declarations';
 import { convertDecoratorsToStatic } from '../decorators-to-static/convert-decorators';
 import { convertStaticToMeta } from '../static-to-meta/visitor';
-import { mockBuildCtx, mockCompilerCtx, mockConfig, mockStencilSystem } from '@stencil/core/testing';
+import { mockBuildCtx, mockCompilerCtx, mockConfig } from '@stencil/core/testing';
 import ts from 'typescript';
 import { updateModule } from '../static-to-meta/parse-static';
 import { getScriptTarget } from '../transform-utils';
 
+/**
+ * Testing utility for transpiling provided string containing valid Stencil code
+ * @param input the code to transpile
+ * @param config a Stencil configuration to apply during the transpilation
+ * @param compilerCtx a compiler context to use in the transpilation process
+ * @param beforeTransformers TypeScript transformers that should be applied before the code is emitted
+ * @param afterTransformers TypeScript transformers that should be applied after the code is emitted
+ * @returns the result of the transpilation step
+ */
 export function transpileModule(
   input: string,
   config?: d.Config,
   compilerCtx?: d.CompilerCtx,
-  sys?: d.CompilerSystem,
```

Comment:
any idea why eslint didn't pick this up as an unused variable?

Reply:
That's a good question 🤔 

I can trigger other lint errors (like our `@param` JSDoc ones) in this file, which tells me the linter is picking up this rule. 

I created a [mini-sample in the ESLint playground](https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gYmFyKGJheikge1xufVxuXG5mdW5jdGlvbiBiYXIyKGJheikge1xuICBiYXogPSBiYXogfHwge307XG59XG52YXIgZm9vID0gNTtcbmJhcigpO1xuYmFyMihmb28pOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWVsc2UtaWYiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWltcG9ydC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZXR0ZXItcmV0dXJuIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyfSwiZW52Ijp7fX19) that shows that the error _should_ be happening (confirming our understanding of the rule itself)
```
```javascript
// both of these functions error in the playground
function bar(baz) {
}

function bar2(baz) {
  baz = baz || {};
}
var foo = 5;
bar();
bar2(foo);
```

I commented out our configuration for `'@typescript-eslint/no-unused-vars'` _just_ in case, and that didn't yield any issues.  So my best guess is either
1. We need to tweak our `'@typescript-eslint/no-unused-vars'` configuration in some way
    1. The rule does suggest turning off the ESLint version, which we don't do today. I added it back locally and couldn't get it to fire on a new (unused) parameter for the function
3. There's an issue (or maybe by design) that no-unused-vars in TS handles this case differently

Tl;dr :shrug:

Reply:
hmmm well sometimes things are just mysteries 🤷‍♀️